purchase_triple_discount/models/purchase_order.py

```diff
@@ -32,11 +38,35 @@ def _compute_amount(self):
          'Discount 3 must be lower than 100%.'),
     ]
 
-    def _get_discounted_price_unit(self):
-        price_unit = super(
-            PurchaseOrderLine, self)._get_discounted_price_unit()
-        if self.discount2:
-            price_unit *= (1 - self.discount2 / 100.0)
-        if self.discount3:
-            price_unit *= (1 - self.discount3 / 100.0)
-        return price_unit
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    @api.model
+    def _prepare_inv_line(self, account_id, line):
+        res = super(PurchaseOrder, self)._prepare_inv_line(account_id, line)
+        res.update({
+            'discount2': line.discount2,
+            'discount3': line.discount3,
+        })
+        return res
+
+    @api.model
+    def _prepare_order_line_move(
+            self, order, order_line, picking_id, group_id):
+        res = super(PurchaseOrder, self)._prepare_order_line_move(
+            order, order_line, picking_id, group_id)
+        for vals in res:
+            vals['price_unit'] = (vals.get('price_unit', 0.0) *
+                                  (1 - (order_line.discount2 / 100)) *
+                                  (1 - (order_line.discount3 / 100)))
+        return res
+
+#    def _get_discounted_price_unit(self):
```

Comment:
Remove this

Reply:
done.

Reply:
Note : should be done upstream, as it is a backport.